pkg/storage/pvc/pvc.go

```diff
@@ -0,0 +1,161 @@
+package pvc
+
+import (
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreset "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+)
+
+const (
+	rootDirectory    = "/registry"
+	randomSecretSize = 32
+)
+
+type driver struct {
+	ClaimName string
+	Namespace string
+	Config    *imageregistryv1.ImageRegistryConfigStoragePVC
+	Client    *coreset.CoreV1Client
+}
+
+func NewDriver(c *imageregistryv1.ImageRegistryConfigStoragePVC) (*driver, error) {
+	namespace, err := regopclient.GetWatchNamespace()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get watch namespace: %s", err)
+	}
+
+	kubeconfig, err := regopclient.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := coreset.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &driver{
+		ClaimName: fmt.Sprintf("%s-storage", imageregistryv1.ImageRegistryName),
```

Comment:
how does a user customize the PVC so they can control the PV they get?

Reply:
@bparees Before that, you told me not to generate this name. If we do not generate a name, then it is under our control in our namespace. User can customize only size.